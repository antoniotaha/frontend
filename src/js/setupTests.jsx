import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import httpAdapter from 'axios/lib/adapters/http'
import axios from 'axios'

export function shallowRenderAndMatch(componentJsx) {
	const wrapper = shallow(componentJsx)
	expect(wrapper).toMatchSnapshot()
	return wrapper
}

Enzyme.configure({ adapter: new Adapter() })
axios.defaults.adapter = httpAdapter 
