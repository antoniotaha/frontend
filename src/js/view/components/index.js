
export * from './elements/Button'
export * from './elements/Card'
export * from './elements/Image'
export * from './elements/Input'
export * from './elements/Label'
export * from './elements/Pagination'
export * from './elements/TextArea'
export * from './elements/TextInput'
export * from './elements/SuccessModal'


export * from './form/Control'
export * from './form/Field'

export * from './grid/Column'
export * from './grid/Columns'

export * from './layout/Header'
export * from './layout/Conteudo'

