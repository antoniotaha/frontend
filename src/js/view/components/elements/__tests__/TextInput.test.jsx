import * as React from 'react'
import * as C from 'view/components'
import { shallowRenderAndMatch } from 'setupTests'

describe('TextInput', () => {
	it('TextInput sem label', function () {
		shallowRenderAndMatch(<C.TextInput />)
	})

	it('TextInput com label', function () {
		shallowRenderAndMatch(<C.TextInput label='Teste' placeholder='Teste' />)
	})
})
