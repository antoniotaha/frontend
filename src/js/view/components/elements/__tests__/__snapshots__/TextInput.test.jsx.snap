// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput TextInput com label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput
    label="Teste"
    placeholder="Teste"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Input
        label="Teste"
        name={undefined}
        placeholder="Teste"
        type="text"
/>,
      "label": "Teste",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "label": "Teste",
        "name": undefined,
        "placeholder": "Teste",
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Input
          label="Teste"
          name={undefined}
          placeholder="Teste"
          type="text"
/>,
        "label": "Teste",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "label": "Teste",
          "name": undefined,
          "placeholder": "Teste",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextInput TextInput sem label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <Input
        name={undefined}
        placeholder={undefined}
        type="text"
/>,
      "label": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "name": undefined,
        "placeholder": undefined,
        "type": "text",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <Input
          name={undefined}
          placeholder={undefined}
          type="text"
/>,
        "label": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "name": undefined,
          "placeholder": undefined,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
